algorithms/unit_tests/TestStdAlgorithmsTeamAdjacentDifference.cpp

```diff
@@ -0,0 +1,206 @@
+        break;
+      }
+
+      case 3: {
+        KE::adjacent_difference(member, myRowView, myDiffView, m_binaryOp);
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(std::size_t numTeams, std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level adjacent_difference
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  auto [dataView, cloneOfDataViewBeforeOp_h] =
+      create_random_view_and_host_clone(
+          LayoutTag{}, numTeams, numCols,
+          Kokkos::pair{ValueType(5), ValueType(523)}, "dataView");
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // to verify that things work, each team stores the result
+  // of its adjacent_difference call, and then we check
+  // that these match what we expect
+  Kokkos::View<ValueType**> diffsView("diffsView", numTeams, numCols);
+
+  // use CTAD for functor
+  TestFunctorA fnc(dataView, diffsView, apiId, PlusFunctor<ValueType>{});
+  Kokkos::parallel_for(policy, fnc);
+
+  // -----------------------------------------------
+  // check
+  // -----------------------------------------------
+  auto dataView_h = create_mirror_view(Kokkos::HostSpace(), dataView);
+  Kokkos::deep_copy(dataView_h, dataView);
+  auto diffsView_h = create_mirror_view(Kokkos::HostSpace(), diffsView);
+  Kokkos::deep_copy(diffsView_h, diffsView);
+
+  Kokkos::View<ValueType**> diffsView_gold_h("diffsView_gold_h", numTeams,
+                                             numCols);
+
+  for (std::size_t i = 0; i < diffsView_h.extent(0); ++i) {
+    auto dataViewRow_h = Kokkos::subview(dataView_h, i, Kokkos::ALL());
+    auto diffsViewRow_gold_h =
+        Kokkos::subview(diffsView_gold_h, i, Kokkos::ALL());
+
+    switch (apiId) {
+      case 0:
+      case 2: {
+        std::adjacent_difference(KE::begin(dataViewRow_h),
+                                 KE::end(dataViewRow_h),
+                                 KE::begin(diffsViewRow_gold_h));
+        break;
+      }
+
+      case 1:
+      case 3: {
+        std::adjacent_difference(
+            KE::begin(dataViewRow_h), KE::end(dataViewRow_h),
+            KE::begin(diffsViewRow_gold_h), PlusFunctor<ValueType>{});
+      }
+    }
+
+    auto diffsViewRow_h = Kokkos::subview(diffsView_h, i, Kokkos::ALL());
+    for (std::size_t j = 0; j < diffsViewRow_gold_h.extent(0); ++j) {
+      EXPECT_FLOAT_EQ(diffsViewRow_gold_h(j), diffsViewRow_h(j));
```

Comment:
also, you use float_eq but the test is run for double, int and unsigned

Reply:
Done.